### PR TITLE
UI: Handle error from errorResponseWithStatusCode helper

### PR DESCRIPTION
**Pull request by hashishaw**
Usually when the Vault API returns an error, the payload looks something like: 
```
{ 
errors: ['message here']
}
```
But when using the [RespondWithStatusCode](https://github.com/hashicorp/vault/blob/e55c18ed1299e0d36b88e603fa9f12adaf8e75dc/sdk/logical/response.go) helper the error response looks like:
```
{
 data: { error: 'message here' }
}
```
When this happens the error format is not recognized by Ember and so the user would see something vague and unhelpful like: 
The dreaded Error [Object object] message

This PR updates the application adapter so it will properly handle the RespondWithStatusCode error format.

### CODE REVIEW

File: ui/tests/acceptance/saml-auth-method-test.js [hunk 1]
```diff
@@ -145,6 +145,26 @@ module('Acceptance | enterprise saml auth method', function (hooks) {
     await click('[data-test-auth-submit]');
   });
 
+  test('it should render API error when it comes from data block', async function (assert) {
```
Comment (hashishaw) [inline ui/tests/acceptance/saml-auth-method-test.js:148]: This use case is what prompted the fix, but if we backport this to earlier versions perhaps we add this test in a different PR (or just remove from the backport?)

Comment (kiannaquach) [reply ui/tests/acceptance/saml-auth-method-test.js:148]: I think it makes sense to add it to a different PR :D

File: ui/app/adapters/application.js [hunk 2]
```diff
@@ -119,10 +119,18 @@ export default RESTAdapter.extend({
 
   handleResponse(status, headers, payload, requestData) {
     const returnVal = this._super(...arguments);
-    // ember data errors don't have the status code, so we add it here
     if (returnVal instanceof AdapterError) {
+      // ember data errors don't have the status code, so we add it here
       set(returnVal, 'httpStatus', status);
       set(returnVal, 'path', requestData.url);
+      // Most of the time when the Vault API returns an error, the payload looks like:
```
Comment (kiannaquach) [inline ui/app/adapters/application.js:126]: thanks for adding the comment!!

**Review by kiannaquach — APPROVED** [review]
🎊

**Pull request merged**